### PR TITLE
Ugly attempt to support OIDs with UUIDs, 2.25.xxx

### PR DESCRIPTION
**Pull request by lokkilok**
Almost certainly not the best way to to this, and for sure not very efficient, but it seems to work. Perhaps can serve as an inspiration for a proper fix.

### CODE REVIEW

**Comment by rmhrisk** [comment]
1?

**Comment by lokkilok** [comment]
As far as I can see both option 1 and 2 could be ok for pretty much anybody who would need this support for OIDs with UUIDs. With option 2 any code that needs this support would have to require that the browser/environment supports BigInts. I guess that this is not too much asked....

**Comment by microshine** [comment]
https://github.com/zloirock/core-js#missing-polyfills
> ES BigInt can't be polyfilled since it requires changes in the behavior of operators, you could find more info here. You could try to use JSBI.

Looks like it's impossible to do the 1st case

File: src/asn1.js [hunk 1]
```diff
@@ -2825,7 +2825,22 @@ class LocalSidValueBlock extends HexBlock(LocalBaseBlock)
 
 		return (inputOffset + this.blockLength);
 	}
-	//**********************************************************************************
+//**********************************************************************************
+	/**
+   * Save a BigInt value immediately as an array of octects.
+   */
+ set valueBigInt(value) {
```
Comment (microshine) [inline src/asn1.js:2832]: Let's add an assert function like
```js
function assertBigInt() {
  if (typeof BigInt === "undefined") {
    throw new Error("BigInt is not defined. Your environment doesn't implement BigInt.")
  }
}
```

and use it here

```suggestion
	set valueBigInt(value)
	{
		assertBigInt();
```

Comment (microshine) [reply src/asn1.js:2832]: And please format your changes. Use TAB instead of SPACE for padding

File: src/asn1.js [hunk 2]
```diff
@@ -3076,9 +3091,13 @@ class LocalObjectIdentifierValueBlock extends ValueBlock
 			else
 			{
 				const sidBlock = new LocalSidValueBlock();
-				sidBlock.valueDec = parseInt(sid, 10);
-				if(isNaN(sidBlock.valueDec))
-					return true;
+        let sidValue = BigInt(sid)
```
Comment (microshine) [inline src/asn1.js:3094]:
```suggestion
        assertBigInt();
        const sidValue = BigInt(sid);
```

**Review by microshine — CHANGES_REQUESTED** [review]

**Comment by rmhrisk** [comment]
> https://github.com/zloirock/core-js#missing-polyfills
> 
> > ES BigInt can't be polyfilled since it requires changes in the behavior of operators, you could find more info here. You could try to use JSBI.
> 
> Looks like it's impossible to do the 1st case
Sounds like 2 is the winner?

**Comment by lokkilok** [comment]
Added the suggestions, thanks! However, I now noted that the code already has a reference to BigInt in the code, in the toString() method of the INTEGER block. In other words the ASN1.js library was already relying on BigInt support....

**Comment by lokkilok** [comment]
@microshine: now that I more or less incorporated the changes you requested (albeit not verbatim) when do you think this will be merged?

**Review by microshine — APPROVED** [review]

**Comment by microshine** [comment]
@lokkilok Thank you for the work you did.

I've published the new version `asn1js@2.2.0`. Please try it

**Comment by lokkilok** [comment]
Lightly tested and seemed ok! Thanks!

**Pull request merged**